### PR TITLE
Marriage testcase #1

### PR DESCRIPTION
**Pull request by Eezi**

### CODE REVIEW

File: e2e/testcases/marriage/1-marriage-event-declaration.spec.ts [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+  test('1.1. Navigate to the event declaration page', async ({ page }) => {
+    await page.click('#header_new_event')
+    await page.waitForSelector('#continue')
+  })
+
+  test('Testcase 1', async ({ page }) => {
+    await page.click('#header_new_event')
+
+    await test.step('1.1.1 Validate the contents of the event type page', async () => {
+      await expect(page.locator('#select_birth_event')).toBeVisible()
```
Comment (jamil314) [inline e2e/testcases/marriage/1-marriage-event-declaration.spec.ts:20]: User-facing attributes like `label` or `text` should be preferred over css selectors when possible
[prefer-user-facing-attributes-to-xpath-or-css-selectors](https://playwright.dev/docs/best-practices#prefer-user-facing-attributes-to-xpath-or-css-selectors)

File: e2e/testcases/marriage/1-marriage-event-declaration.spec.ts [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        page.locator('#form_section_id_documents-view-group')
+      ).toBeVisible()
+    })
+    await test.step('1.9. Validate Supporting document page', async () => {
+      await validateSectionButtons(page)
+      await page.click('#next_section')
+      await expect(page.locator('#review_header')).toBeVisible()
+    })
+  })
+  test('Testcase 1 Save and exit button', async ({ page }) => {
```
Comment (jamil314) [inline e2e/testcases/marriage/1-marriage-event-declaration.spec.ts]:
```suggestion
  test('1.11 Validate save and exit button', async ({ page }) => {
```

File: e2e/testcases/marriage/1-marriage-event-declaration.spec.ts [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+    })
+
+    await test.step('1.11.3. Confirm "Save & Exit" button', async () => {
+      await page.click('#save-exit-btn')
+      await page.click('#confirm_save_exit')
+      await expect(page.locator('#content-name')).toBeVisible()
+      await expect(page.locator('#name_0')).toBeVisible()
+    })
+  })
+  test('Testcase 1 Exit button', async ({ page }) => {
```
Comment (jamil314) [inline e2e/testcases/marriage/1-marriage-event-declaration.spec.ts]:
```suggestion
  test('1.12 Validate exit button', async ({ page }) => {
```

File: e2e/testcases/marriage/1-marriage-event-declaration.spec.ts [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    })
+    await test.step('1.12.3. Confirm "Exit" button', async () => {
+      await page.getByText('Exit', { exact: true }).click()
+      await page.click('#confirm_save_without_exit')
+      await expect(page.locator('#content-name')).toBeVisible()
+      await expect(page.locator('#no-record')).toBeVisible()
+    })
+  })
+
+  test('Testcase 1 Three dot menu button', async ({ page }) => {
```
Comment (jamil314) [inline e2e/testcases/marriage/1-marriage-event-declaration.spec.ts]:
```suggestion
  test('1.13 Validate three dot menu button', async ({ page }) => {
```

**Review by jamil314 — CHANGES_REQUESTED** [review]
Branch name needs to be `e2e/marriage` for e2e workflows to trigger

File: e2e/testcases/marriage/1-marriage-event-declaration.spec.ts (hunk 1 above)
Comment (Eezi) [reply e2e/testcases/marriage/1-marriage-event-declaration.spec.ts:20]: Nice to know! Should change those id selectors to text or label?

Comment (jamil314) [reply e2e/testcases/marriage/1-marriage-event-declaration.spec.ts:20]: Yes, use label / text when possible.

Comment (Eezi) [reply e2e/testcases/marriage/1-marriage-event-declaration.spec.ts:20]: I opened new pr from this and named it correctly so it triggers the e2e workflows and changed all the id selectors to text or label selectors.
Here is the new pr: https://github.com/opencrvs/opencrvs-farajaland/pull/1013

**Pull request closed without merging**